pyvrp/cpp/diversity/broken_pairs_distance.cpp

```diff
@@ -13,13 +13,19 @@ double pyvrp::diversity::brokenPairsDistance(pyvrp::Solution const &first,
 
     for (size_t j = 1; j <= numClients; j++)
     {
-        auto const [fPred, fSucc] = fNeighbours[j];
-        auto const [sPred, sSucc] = sNeighbours[j];
+        if (fNeighbours[j].has_value() && sNeighbours[j].has_value())
+        {
+            // Both clients are assigned, compare neighbours
+            auto const [fPred, fSucc] = fNeighbours[j].value();
+            auto const [sPred, sSucc] = sNeighbours[j].value();
 
-        // An edge pair (fPred, j) or (j, fSucc) from the first solution is
-        // broken if it is not in the second solution.
-        numBrokenPairs += fSucc != sSucc;
-        numBrokenPairs += fPred != sPred;
+            // An edge pair (fPred, j) or (j, fSucc) from the first solution is
+            // broken if it is not in the second solution.
+            numBrokenPairs += fSucc != sSucc;
+            numBrokenPairs += fPred != sPred;
+        }
+        else if (fNeighbours[j].has_value() || sNeighbours[j].has_value())
+            numBrokenPairs += 2;  // One assigned, other not so 2 broken edges
```

Comment:
This could be a much smaller change, right?

```cpp
std::pair<size_t, size_t> constexpr unassigned = {numClients + 1, numClients + 1};

...

auto const [fPred, fSucc] = fNeighbours[j].value_or(unassigned);
auto const [sPred, sSucc] = ...;
```